football_dictionaries/assignment_3.py

```diff
@@ -1,2 +1,28 @@
+from football_dictionaries.squads_data import SQUADS_DATA
+
+
+def player_list_to_dict(player_list):
+    return{'number': player_list[0],
+           'position': player_list[1],
+           'name': player_list[2],
+           'date_of_birth': player_list[3],
+           'caps': player_list[4],
+           'club': player_list[5],
+           'country': player_list[6],
+           'club_country': player_list[7],
+           'year': player_list[8]
+           }
+
+
 def players_by_country_and_position(squads_list):
-    pass
+    result = {}
+    for player_list in squads_list:
+        player = player_list_to_dict(player_list)
+        country = player['country']
+        position = player['position']
+        if country not in result:
```

Comment:
Hey D,

Here you can use the setdefault method aswell.

I recommend you to check it out. I went over this exercise with one of the mentors, because I needed help to get my head around it. It was really useful to understand the setdefault method and how it can be used.

Reply:
Hello Adriana, 
Thanks! a bunch for this. Really appreciate your feedback. 
The assignments in this project really threw me off and finally for assignment 3, I was so lost, I ended up following Santiago's video and pretty much mimicking the code. 
I understood the way the code works but just could not come up with the solution on my own (esp. nested grouping). 
I will review setdefault method & 10/10 to you if I could rate your review. 

Regards,
Dan